streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -262,7 +265,7 @@ boolean maybeResumeSuspendedTask(final TaskId taskId, final Set<TopicPartition>
                     suspended.remove(taskId);
                     throw e;
                 }
-                transitionToRunning(task);
+                transitionToRunning(task, readyPartitions);
```

Comment:
Why do we maintain a local `HashSet` that is never read?

Reply:
just to avoid creating one each time through the loop

Reply:
Can't we pass a `null` and add a check with `transitionToRunning()` -- it's not a critical code path...

Reply:
@dguy What about this?

Reply:
not really a fan of passing null around

Reply:
Can you at least add a comment or rename the variable to `dummyToAvoidNull` (or similar)? Otherwise this code is confusing.

Reply:
Actually I'm thinking about the line above:

```
task.partitions().equals(partitions)
```

Could this ever be false assuming there is no bugs, or are we adding that check just to-be-sure? One wild thought I have if otherwise, is the following: blindly call `transitionToRunning` without the above check with an empty set, and after it returns do an after-check with the `partitions` set, and if they are not equal treat it as a fatal error instead of logging as TRACE. WDYT?

Reply:
The partitions that are only added to the set if they task has state stores, so this could be empty. So we can't use that. However, i think you are correct in that only a bug should result in the two partitions sets not being equal, so maybe we should throw an exception?

Reply:
That sounds good in general. Not sure which exception we should throw though. Maybe a generic `StreamsException` ?